benchmark.py

```
@@ -0,0 +1,318 @@
+print("─" * 80)
+
+for N in [50, 100, 200]:
+    coords = random_coords(N)
+    n_pairs = N * (N - 1) // 2
+    print(f"\n  N = {N} points ({n_pairs:,} unique pairs)")
+    print(f"  {'Library':<30} {'Time':>12}  {'vs Geopy':>10}")
+    print(f"  {'─'*30} {'─'*12}  {'─'*10}")
+
+    def run_geopy_matrix():
+        dists = []
+        for i in range(len(coords)):
```

Ruff (0.15.1)

[warning] Function definition does not bind loop variable `coords`

(B023)